11_FFTBloom/app_resources/kernel_spectrum_normalize.hlsl

```diff
@@ -0,0 +1,124 @@
+#include "common.hlsl"
+#include "nbl/builtin/hlsl/colorspace/encodeCIEXYZ.hlsl"
+
+/*
+ * Remember we have these defines:
+ * _NBL_HLSL_WORKGROUP_SIZE_
+ * ELEMENTS_PER_THREAD
+ * KERNEL_SCALE
+ * scalar_t
+ * FORMAT
+*/
+
+#define IMAGE_SIDE_LENGTH (_NBL_HLSL_WORKGROUP_SIZE_ * ELEMENTS_PER_THREAD)
+
+// Users MUST define this method for FFT to work
+uint32_t3 glsl::gl_WorkGroupSize() { return uint32_t3(_NBL_HLSL_WORKGROUP_SIZE_, 1, 1); }
+
+[[vk::binding(0, 0)]] [[vk::image_format( FORMAT )]] RWTexture2D<float32_t2> kernelChannels[CHANNELS];
+
+// ---------------------------------------------------- Utils ---------------------------------------------------------
+uint64_t rowMajorOffset(uint32_t x, uint32_t y)
+{
+	return y * IMAGE_SIDE_LENGTH | x;
+}
```

Comment:
you know that as long as each invocation touches the same image texel, you can read and write to the same image?